cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/InMemoryProgramRunDispatcher.java

```diff
@@ -477,6 +480,11 @@ private ProgramOptions updateProgramOptions(ArtifactId artifactId, ProgramId pro
                                     new BasicArguments(userArguments), options.isDebug());
   }
 
+  private static void hashArtifactId(Hasher hasher, ArtifactId artifactId) {
+    hasher.putString(String.format("%s.%s.%s.%s", artifactId.getParent().toString(), artifactId.getArtifact(),
```

Comment:
You can call `putString`/`putInt` multiple times instead of using `String.format`.